dynamic/msgregistry/message_registry.go

```diff
@@ -22,6 +22,42 @@ import (
 
 const googleApisDomain = "type.googleapis.com"
 
+// ErrUnexpectedEnum is returned if the URL that was requested
+// resolved to an enum instead of a message.
+type ErrUnexpectedEnum struct {
+	URL                string
+	FullyQualifiedName string
```

Comment:
This field is redundant with the one above, right? IIRC, the fully-qualified name will always be the last path element of the URL. Maybe we can get by omitting this?

Reply:
It is indeed, but I included it since it may not always be obvious to do the reverse transformation. I will remove it if you think it's unnecessary.